x-pack/plugins/security_solution/common/endpoint/generate_data.ts

```diff
@@ -354,16 +367,93 @@ export function getTreeOptionsWithDef(options?: TreeOptions): TreeOptionDefaults
   };
 }
 
+const metadataDefaultDataStream = {
+  type: 'metrics',
+  dataset: 'endpoint.metadata',
+  namespace: 'default',
+};
+
+const policyDefaultDataStream = {
+  type: 'metrics',
+  dataset: 'endpoint.policy',
+  namespace: 'default',
+};
+
+const eventsDefaultDataStream = {
+  type: 'logs',
+  dataset: 'endpoint.events.process',
+  namespace: 'default',
+};
+
+const alertsDefaultDataStream = {
+  type: 'logs',
+  dataset: 'endpoint.alerts',
+  namespace: 'default',
+};
+
+interface EndpointDocGeneratorParams {
+  seed?: string | seedrandom.prng;
+  metadataDataStream?: DataStream;
+  policyDataStream?: DataStream;
+  eventsDataStream?: DataStream;
+  alertsDataStream?: DataStream;
+}
+
+function createDataStreamField(ds: DataStream) {
+  return { data_stream: ds };
+}
+
 export class EndpointDocGenerator {
   commonInfo: HostInfo;
   random: seedrandom.prng;
   sequence: number = 0;
-  constructor(seed: string | seedrandom.prng = Math.random().toString()) {
-    if (typeof seed === 'string') {
-      this.random = seedrandom(seed);
+  metadataDataStream: DataStreamField = createDataStreamField(metadataDefaultDataStream);
+  policyDataStream: DataStreamField = createDataStreamField(policyDefaultDataStream);
+  eventsDataStream: DataStreamField = createDataStreamField(eventsDefaultDataStream);
+  alertsDataStream: DataStreamField = createDataStreamField(alertsDefaultDataStream);
+  /**
+   * The EndpointDocGenerator parameters
+   *
+   * @param seed either a string to seed the random number generator or a random number generator function
+   * @param metadataDataStream the values to populate the data_stream fields when generating metadata documents
+   * @param policyDataStream the values to populate the data_stream fields when generating policy documents
+   * @param eventsDataStream the values to populate the data_stream fields when generating event documents
+   * @param alertsDataStream the values to populate the data_stream fields when generating alert documents
+   */
+  constructor(
+    options: string | seedrandom.prng | EndpointDocGeneratorParams = {
```

Comment:
With `seed` as a field inside `EndpointDocGeneratorParams` I don't think we should allow `options` to be `string` or `seedrandom.prng` - we should always use `options.seed` or add the data stream params as a separate parameter

Reply:
yeah that makes sense. I was just being lazy and trying to avoid changing all the places we use `EndpointDocGenerator` 😄  I'll update them